.github/workflows/determine-workflow-runner.yml

```diff
@@ -0,0 +1,65 @@
+name: Determine Workflow Runner group
+
+on:
+  workflow_call:
+    inputs:
+      default_runner:
+        description: The runner type that is used by the calling workflow by default
+        required: true
+        type: string
+      large_runner_group_name:
+        description: The name of the runner group that should be used for large jobs
+        required: false
+        type: string
+        default: 'pl-4-core-large-runner'
+      force_large_runner:
+        description: Whether to force the use of a large runner
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      runner_group:
+        description: The runner all subsequent jobs within the calling workflow should run on
+        value: >-
+          ${{
+            jobs.determine_workflow_runner.outputs.runner_group ||
+            (
+              inputs.force_large_runner &&
+              inputs.large_runner_group_name ||
+              inputs.default_runner
+            )
+          }}
+
+jobs:
+  determine_workflow_runner:
```

Comment:
Shouldn't this job get skipped if `force_large_runner` is true?

Reply:
Oh wait, so the way you've set it up is that we choose the output of `determine_workflow_runner` if it finishes, and if not, only then we care about `force_large_runner.

Reply:
Great catch! This is a logic bug, going to address it

Reply:
If force large runner is true, we always want large runner

Reply:
Oh wait, no logic bug @mudit2812 ... the `determine_workflow_runner` either outputs the name of the large runner group OR doesn't output anything at all.

So it's either the larger runner group from the workflow OR the force large runner logic.

Reply:
So this is what I understand. Let me know if this is indeed the behaviour we want.
* If the job is used by a pull request, then we always use the output of `determine_workflow_runner` as the source of truth. In this case, we use the large runners iff:
  * The PR has the `urgent` label, or
  * The PR is targeting an RC branch.
* If the job trigger is not a pull request, _only then_ we use `force_large_runner` to determine whether or not to use the large runner or default runner.

Reply:
* If the job is used by a pull request, then we always use the output of `determine_workflow_runner` as the source of truth. In this case, we use the large runners if:
  * The PR has the `urgent` label, or :heavy_check_mark: 
  * The PR is targeting an RC branch. :x: -- The check is running on the rc branch itself!
* If the job trigger is not a pull request, _only then_ we use `force_large_runner` to determine whether or not to use the large runner or default runner. This is because if the workflow trigger is not a pull request, then the `determine_workflow_runner` job will get skipped. :heavy_check_mark:

Reply:
If a PR is going to be *merged into* rc branch, it would need the `urgent` label, however the checks that'd on the rc branch once the PR is merged would automatically use the large runners

Reply:
Correct me if I'm wrong, but the condition used below to use the large runners below is
```yaml
 if [[ '${{ contains(github.event.pull_request.labels.*.name, 'urgent') }}' == 'true' || ('${{ github.event.pull_request.head.repo.full_name }}' == "$REPO_FULL_NAME" && '${{ github.event.pull_request.head.ref }}' =~ $RC_BRANCH_FORMAT_REGEX) ]]; then
```
Specifically, for the following part
```yaml
('${{ github.event.pull_request.head.repo.full_name }}' == "$REPO_FULL_NAME" && '${{ github.event.pull_request.head.ref }}' =~ $RC_BRANCH_FORMAT_REGEX)
```
Isn't this saying "if the event is a pull request and the head of the PR is the RC branch, then use the large runner"?

But what you're saying we want is: "if the event is a _push_ into the RC branch, then use the large runner".

So shouldn't the condition be the following?
```yaml
('${{ github.event.push.repo.full_name }}' == "$REPO_FULL_NAME" && '${{ github.event.push.ref }}' =~ $RC_BRANCH_FORMAT_REGEX)
```

Reply:
Great point @mudit2812 

I have updated the logic so that if a PR has the RC branch as the base, it will automatically use large runner without label;